### PR TITLE
Add test for `PutFollowAction` on a closed index

### PR DESCRIPTION
**Pull request by Tim-Brooks**
This is related to #35975. Currently when an index falls behind a leader
it encounters a fatal exception. This commit adds a test for that
scenario. Additionally, it tests that the user can stop following, close
the follower index, and put follow again. After the indexing is
re-bootstrapped, it will recover the documents it lost in normal
following operations.

### CODE REVIEW

**Review by jasontedor — APPROVED** [review]
This looks good. Thanks for doing this. In a follow-up, can we also add some documentation for this situation?

File: x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/IndexFollowingIT.java [hunk 1]
```diff
@@ -947,6 +954,97 @@ public void testUpdateAnalysisLeaderIndexSettings() throws Exception {
...
+        followerClient().execute(ResumeFollowAction.INSTANCE, followRequest.getFollowRequest()).get();
+
+        assertBusy(() -> {
+            List<ShardFollowNodeTaskStatus> statuses = getFollowTaskStatuses("index2");
+            Set<ResourceNotFoundException> exceptions = statuses.stream()
+                .map(ShardFollowNodeTaskStatus::getFatalException)
+                .filter(Objects::nonNull)
+                .map(ExceptionsHelper::unwrapCause)
+                .filter(e -> e instanceof ResourceNotFoundException)
+                .map(e -> (ResourceNotFoundException) e)
```
Comment (martijnvg) [inline x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/IndexFollowingIT.java]: Maybe also exclude exception that do not have `es.requested_operations_missing` as metadata? Then we are even more sure that the exception is the right one.

**Review by martijnvg — APPROVED** [review]
Left one comment, LGTM otherwise.

**Review by dnhatn — APPROVED** [review]
Left some nits but LGTM.

File: x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/IndexFollowingIT.java [hunk 2]
```diff
@@ -947,6 +954,78 @@ public void testUpdateAnalysisLeaderIndexSettings() throws Exception {
...
+                .filter(e -> {
+                    return e.getMetadataKeys().contains("es.requested_operations_missing");
+                })
+                .collect(Collectors.toSet());
+            assertThat(exceptions.size(), greaterThan(0));
+        });
+
+        pauseFollow("index2");
```
Comment (ywelsch) [inline x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/IndexFollowingIT.java]: what happens if we close before pausing following? What happens if we don't explicitly pause following at all? Let's also check these scenarios.

**Review by ywelsch — CHANGES_REQUESTED** [review]

File: x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/IndexFollowingIT.java (hunk 2 above)
Comment (tbrooks8) [reply x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/IndexFollowingIT.java]: > what happens if we close before pausing following?

It works. We allow closing a follower index. There is a test that checks this behavior `testCloseFollowIndex`. 

> Let's also check these scenarios.

I added a test for the failure scenarios in the restart following case (index still open and following not paused).

**Review by ywelsch — APPROVED** [review]

**Pull request merged**